build/verifyLinks.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will scan all md (markdown) files for bad references.
+# It will look for strings of the form [...](...) and make sure that
+# the (...) points to either a valid file in the source tree or, in the
+# case of it being an http url, it'll make sure we don't get a 404.
+#
+# Usage: verifyLinks.sh [ dir | file ... ]
+# default arg is root of our source tree
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+if [ "$*" != "" ]; then
+  args="$*"
+else
+  args="${REPO_ROOT}"
+fi
+
+mdFiles=$(find "${args}" -name "*.md" | grep -v vendor | grep -v glide)
+
+tmp=/tmp/out${RANDOM}
+
+rm -f /tmp/$tmp*
+for file in ${mdFiles}; do
+  # echo scanning $file
+  dir=$(dirname $file)
+  sed "s/)/)\n/g" < $file | grep "\[.*\](.*)" > ${tmp}1 || continue
+  sed "s/.*\(\[[^\[\]*\]([^()]*)\)/\1/" < ${tmp}1 > ${tmp}2  || continue
+  cat ${tmp}2 | while read line ; do
+    ref=${line#*(}
+	ref=${ref%)*}
+	if [ "${ref:0:4}" == "http" ]; then 
+	  if ! curl --connect-timeout 10 -L ${ref} > /dev/null 2>&1 ; then
+	    echo $file: Can\'t load: url ${ref} | tee -a ${tmp}3
+	  fi
+	  continue
+	fi
+	if [ "${ref:0:1}" == "#" ]; then 
```

Comment:
correct - converting md anchors into the right href url seemed like it might take some investigation so I was going to save that for a follow-on PR.